Packet++/src/SSLHandshake.cpp

```diff
@@ -1216,6 +1216,20 @@ namespace pcpp
 		return result;
 	}
 
+	// --------------------------
+	// SSLx509Certificate methods
+	// --------------------------
+
+	Asn1SequenceRecord* SSLx509Certificate::getRootAsn1Record()
+	{
+		if (m_Asn1Record == nullptr)
```

Comment:
I am not familiar with the protocol, but I wonder what if `m_Asn1Record` has something, do we want to update it?

Reply:
This `SSLx509Certificate` class is "read only", meaning updating the instance is impossible.

This pattern is for lazy loading - only load it if the user asks